### ISSUE TITLE
Delete a simulation should delete any associated taskflows

### ISSUE THREAD

**Issue by cjh1** [Kitware/HPCCloud#238]
In order for simulation result on the cluster to be cleaned up the taskflows should be deleted ....

**Comment by TristanWright** [comment]
Should this be handled on the server side or do it on the front?

**Comment by cjh1** [comment]
The server side has no knowledge of with taskflows are associated with a
sim, so I think it will have to be done on the client.

**Comment by TristanWright** [comment]
Is there any risk in deleting taskflows which are in the terminating state?

**Comment by cjh1** [comment]
yes, they should only be deleted if they are in the 'created' or
'terminated' or 'error' states

**Comment by TristanWright** [comment]
Maybe we can use the taskflow monitor to figure out when the tasks are terminated and then to trigger a delete?

How is this conveyed on the front? Does the user have to wait for each tf to terminate and delete to see the simulation removed from the list?

**Comment by cjh1** [comment]
So currently when you issue a:

```
DELETE /taskflows/<id>
```

The call will either return status code of 200 and payload contain the taskflow with a status of 'deleted', in this case the taskflow has been deleted. 

Or a status code of 202 ( indicating that delete request has been initialed ). To check that when the taskflow has been delete you would then have to make calls to:

```
GET /taskflows/<id>
```

When this return 400 you know the taskflow has been removed. 

However, to make things easier for the client. I think I can add a state change server side event for the completion of the delete operation:

```json
{
  "type": "taskflow.status",
  "data": {
    "_id": <taskflow id>,
    "status": "deleted"
  }
}
```

So when a user initials a delete of a simulation the following should happen:

- Mark the simulation as "deleting" in some way.
- Delete all taskflows associated with the simulation.
- Wait the above deleted events.
- Hit the delete endpoint for the simulation

Does this make sense? If so I can add the "deleted" event.

**Comment by TristanWright** [comment]
Closed with #512

**Comment by cjh1** [comment]
@TristanWright Could you raise an enhancement issue, that references this one. We can improve the delete process ( i.e. reporting errors, we currently assume is has succeeded )
